src/main/java/org/gridsuite/network/map/NetworkMapService.java

```diff
@@ -251,28 +254,71 @@ private static ThreeWindingsTransformerMapData toMapData(ThreeWindingsTransforme
         return builder.build();
     }
 
-    public List<SubstationMapData> getSubstations(UUID networkUuid) {
-        Network network = getNetwork(networkUuid);
-        return network.getSubstationStream().map(NetworkMapService::toMapData).collect(Collectors.toList());
+    public List<SubstationMapData> getSubstations(UUID networkUuid, List<String> substationsId) {
+        Network network = getNetwork(networkUuid, substationsId == null ? PreloadingStrategy.COLLECTION : PreloadingStrategy.NONE);
+        if (substationsId == null) {
+            return network.getSubstationStream()
+                .map(NetworkMapService::toMapData).collect(Collectors.toList());
+        } else {
+            List<SubstationMapData> res = new ArrayList<>();
+            substationsId.stream().forEach(id -> res.add(toMapData(network.getSubstation(id))));
+            return res;
+        }
     }
 
-    public List<LineMapData> getLines(UUID networkUuid) {
-        Network network = getNetwork(networkUuid);
-        return network.getLineStream().map(NetworkMapService::toMapData).collect(Collectors.toList());
+    public List<LineMapData> getLines(UUID networkUuid, List<String> substationsId) {
+        Network network = getNetwork(networkUuid, substationsId == null ? PreloadingStrategy.COLLECTION : PreloadingStrategy.NONE);
+        if (substationsId == null) {
+            return network.getLineStream()
+                    .map(NetworkMapService::toMapData).collect(Collectors.toList());
+        } else {
+            Set<LineMapData> res = new HashSet<>();
+            substationsId.stream().forEach(id ->
```

Comment:
Should we index the elements in the database by their substationId to avoid fetching the substations ?

Reply:
And get them all at once to avoid reloading the substation and voltage levels many times ?